### PR TITLE
Fix CMS deprecation warnings in  RecoEgamma subsystem

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- added esConsumes
- converted modules to one::EDAnalyzer

#### PR validation:

Packages compile without issuing a CMS deprecation warning.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoEgamma/EgammaTools/src/HoECalculator.cc [hunk 1]
```diff
@@ -4,15 +4,15 @@
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
 
-HoECalculator::HoECalculator() : theCaloGeom_(nullptr) {}
+//HoECalculator::HoECalculator() : theCaloGeom_(nullptr) {}
```
Comment (clacaputo) [inline RecoEgamma/EgammaTools/src/HoECalculator.cc:7]: why don't remove this line?

Comment (Dr15Jones) [reply RecoEgamma/EgammaTools/src/HoECalculator.cc]: removed

**Comment by clacaputo** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by clacaputo** [comment]
+reconstruction

- technical PR

**Comment by rappoccio** [comment]
+1 

* Technical

**Pull request merged**